backend/src/Designer/Services/Interfaces/ISchemaModelService.cs

```diff
@@ -84,5 +84,15 @@ public interface ISchemaModelService
         /// <param name="cancellationToken">An <see cref="CancellationToken"/> that observes if operation is cancelled.</param>
         /// <returns>Returns the model metadata</returns>
         Task<ModelMetadata> GenerateModelMetadataFromJsonSchema(AltinnRepoEditingContext altinnRepoEditingContext, string relativeFilePath, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Gets the dataType for a given model.
+        /// </summary>
+        Task<DataType> GetModelDataType(string org, string app, string modelId);
+
+        /// <summary>
+        /// Updates the dataType for a given model.
+        /// </summary>
+        Task SetModelDataType(string org, string app, string modelId, DataType dataType);
```

Comment:
```suggestion
        /// Gets the dataType in applicationMetadata for a given data model.
        /// </summary>
        Task<DataType> GetModelDataTypeInApplicationMetadata(string org, string app, string modelId);

        /// <summary>
        /// Updates the dataType in applicationMetadata for a given data model.
        /// </summary>
        Task SetModelDataTypeInApplicationMetadata(string org, string app, string modelId, DataType dataType);
```

Reply:
I don't think i agree with this change, can you elaborate?
From my experience, the service layer exists to create an abstraction layer between the controller and the storage details. Exposing where the data is stored to the controller layer adds (in my opinion unecessary) coupling between the layers which reduces readability and maintainability of the code.

Reply:
I think this makes it more clear what the method actually does before I read the actual implementation. That way we wont by some very wired mistake use this method in places where it is not clear what the side effects are.

Reply:
I think the idea here is that you shouldn't need to worry about these details (in the controller layer). If we in the future change where the DataType object is stored, why should the controller have to change? I think naming it like suggested blur the lines of what part of the code should bear what responsibility, and can cause more problems.

Reply:
Hm, guess it comes down to preferences then. No biggie for me, so lets keep it as is.